tests/ttnn/unit_tests/operations/test_silu.py

```diff
@@ -0,0 +1,159 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import math
+from typing import Union, Tuple
+from loguru import logger
+import torch
+import torch.nn as nn
+
+# import tt_lib
+import tt_lib as ttl
+import ttnn
+
+from tests.ttnn.utils_for_testing import assert_with_pcc, check_with_pcc_without_tensor_printout
+from models.utility_functions import skip_for_wormhole_b0
+from tt_lib.utils import (
+    _nearest_y,
+)
+
+TILE_WIDTH = 32
+
+
+def get_shard_grid_from_num_cores(ncores: Union[int, Tuple[int, int]]) -> ttnn.experimental.tensor.CoreRangeSet:
+    max_grid_size = (9, 12)  ## (y, x)
```

Comment:
max grid for WH is different. Please use an api to get max grid given a device object. There is definitely one in ttlib and I am not sure if it has been ported to TTNN. @tapspatel do you know the ttnn api to get max grid size from device?

Reply:
you can get device from the tensor via `input_tensor.device()`
pass that into your function
query via `max_grid_size = (8, 8) if device.arch() == ttl.device.Arch.WORMHOLE_B0 else (9, 12)  ## (y, x)`
check `ttnn/ttnn/operations/others.py` function `def upsample` I've added this change there

Reply:
alternatively, you can also use device object `device.arch() == ttl.device.Arch.WORMHOLE_B0`